### PR TITLE
ENH: update backend tests

### PR DESCRIPTION
**Pull request by samir-nasibli**
## Description
* redesigned backend testing

## TODO
* all todos from `dpnp/backend/tests/test_random.cpp`
* some fix for https://github.com/IntelPython/dpnp/issues/551

### CODE REVIEW

**Comment by samir-nasibli** [comment]
@shssf CI old fails seem are not related with this PR changes.
All needed changes for this PR logic was added and discussed.
We can merge it as is. On other PR I will try to build and run this test on new azure-pipelines job.

File: dpnp/backend/tests/test_main.cpp [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+// THE POSSIBILITY OF SUCH DAMAGE.
+//*****************************************************************************
+
+#include "gtest/gtest.h"
+
+#include <dpnp_iface.hpp>
+
+// TODO add namespace
+// will added for test_commons
+class DPNPTestEnvironment : public testing::Environment
```
Comment (shssf) [inline dpnp/backend/tests/test_main.cpp:32]:
```suggestion
class DPNPCTestEnvironment : public testing::Environment

```

File: dpnp/backend/tests/test_main.cpp [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+public:
+    DPNPTestEnvironment(QueueOptions selection)
+    {
+        this->selection = selection;
+    }
+    void SetUp() override
+    {
+        // TODO update print
+        std::cout << "starting new env" << std::endl << std::endl;
+        dpnp_queue_initialize_c(selection);
```
Comment (shssf) [inline dpnp/backend/tests/test_main.cpp:43]: this is not needed here becasue it is already implemented and work inside the backend. The queue will be initialized on first call

File: dpnp/backend/tests/test_main.cpp [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+    }
+    void TearDown() override
+    {
+    }
+
+private:
+    QueueOptions selection;
```
Comment (shssf) [inline dpnp/backend/tests/test_main.cpp:50]: control of the queue managed by the backend or dpctl. We don't need one more extra control place

**Review by shssf — APPROVED** [review]

File: dpnp/backend/tests/test_random.cpp [hunk 4]
```diff
@@ -64,115 +113,75 @@ bool check_statistics(_DataType* r, double tM, double tD, double tQ, size_t size
...
+    dpnp_rng_srand_c(seed);
+    dpnp_rng_beta_c<double>(result2, a, b, size);
 
-        for (size_t i = 0; i < size; ++i)
-        {
-            EXPECT_NEAR(result1[i], result2[i], 0.004);
-        }
+    for (size_t i = 0; i < size; ++i)
+    {
+        EXPECT_NEAR(result1[i], result2[i], 0.004);
```
Comment (Alexander-Makaryev) [inline dpnp/backend/tests/test_random.cpp:130]: why you are expecting close but not the same values?
looks like values should be the same if you are using the same seed.

File: dpnp/backend/tests/test_random.cpp [hunk 5]
```diff
@@ -64,115 +113,75 @@ bool check_statistics(_DataType* r, double tM, double tD, double tQ, size_t size
...
-        dpnp_rng_srand_c(seed);
-        dpnp_rng_f_c<double>(result2, dfnum, dfden, size);
-
-        for (size_t i = 0; i < size; ++i)
-        {
-            EXPECT_NEAR(result1[i], result2[i], 0.004);
-        }
+    for (size_t i = 0; i < size; ++i)
+    {
+        EXPECT_NEAR(result1[i], result2[i], 0.004);
```
Comment (Alexander-Makaryev) [inline dpnp/backend/tests/test_random.cpp:148]: The same comment as above

File: dpnp/backend/tests/test_random.cpp [hunk 6]
```diff
@@ -64,115 +113,75 @@ bool check_statistics(_DataType* r, double tM, double tD, double tQ, size_t size
...
+    dpnp_rng_srand_c(seed);
+    dpnp_rng_normal_c<double>(result2, loc, scale, size);
 
-        for (size_t i = 0; i < size; ++i)
-        {
-            EXPECT_NEAR(result1[i], result2[i], 0.004);
-        }
+    for (size_t i = 0; i < size; ++i)
+    {
+        EXPECT_NEAR(result1[i], result2[i], 0.004);
```
Comment (Alexander-Makaryev) [inline dpnp/backend/tests/test_random.cpp:166]: The same comment as above

File: dpnp/backend/tests/test_random.cpp [hunk 7]
```diff
@@ -64,115 +113,75 @@ bool check_statistics(_DataType* r, double tM, double tD, double tQ, size_t size
...
+    dpnp_rng_srand_c(seed);
+    dpnp_rng_uniform_c<double>(result2, low, high, size);
 
-        for (size_t i = 0; i < size; ++i)
-        {
-            EXPECT_NEAR(result1[i], result2[i], 0.004);
-        }
+    for (size_t i = 0; i < size; ++i)
+    {
+        EXPECT_NEAR(result1[i], result2[i], 0.004);
```
Comment (Alexander-Makaryev) [inline dpnp/backend/tests/test_random.cpp:184]: The same comment as above

File: dpnp/backend/tests/test_random.cpp (hunk 4 above)
Comment (samir-nasibli) [reply dpnp/backend/tests/test_random.cpp:130]: it is common case for checking floating point values. But you are right it is better check the same values here for the same device calculation.

Comment (samir-nasibli) [reply dpnp/backend/tests/test_random.cpp:130]: @Alexander-Makaryev this changes will be done on refactoring this common tests on other PR. This tests case will be generalized for most random distribution functions.  Needed changes will be done there. I am leaving this PR as is.

File: dpnp/backend/tests/test_random.cpp (hunk 5 above)
Comment (samir-nasibli) [reply dpnp/backend/tests/test_random.cpp:148]: see https://github.com/IntelPython/dpnp/pull/552#discussion_r572694839

**Comment by samir-nasibli** [comment]
@Alexander-Makaryev @shssf Most works for this PR are done. This PR was for restructuring generally backend tests. other comment on https://github.com/IntelPython/dpnp/pull/552#discussion_r572694839

**Review by Alexander-Makaryev — APPROVED** [review]
🔢

**Pull request merged**